src/EnergyPlus/WeatherManager.cc

```diff
@@ -2309,33 +2309,35 @@ namespace WeatherManager {
-        if ((state.dataEnvrn->RunPeriodEnvironment) && (!state.dataGlobal->WarmupFlag)) {
-            int month = state.dataEnvrn->Month;
-            state.dataWaterData->RainFall.MonthlyTotalPrecInWeather.at(month - 1) += state.dataEnvrn->LiquidPrecipitation * 1000.0;
-            if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
-                state.dataWaterData->RainFall.numRainyHoursInWeather.at(month - 1) += 1;
-            }
-        }
 
         // if using epw precipitation, throw warning
         if (state.dataWaterData->RainFall.ModeID == DataWater::RainfallMode::EPWPrecipitation) {
-            if ((state.dataEnvrn->LiquidPrecipitation == 0) && (state.dataEnvrn->IsRain)) {
-                state.dataEnvrn->LiquidPrecipitation = 1.5 / 1000.0;
+            if ((state.dataEnvrn->LiquidPrecipitation == 0) &&
+                // this is the Present Weather Codes field in epw. See column 1-3 in Table 2.16 Weather Codes Field Interpretation
+                (state.dataWeatherManager->TodayIsRain(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay)) &&
+                (state.dataWeatherManager->UseRainValues)) {
+                state.dataEnvrn->LiquidPrecipitation = 1.5 / 1000.0 / state.dataGlobal->TimeStepZone;
                 ShowRecurringWarningErrorAtEnd(state,
                                                "Rain flag is on but precipitation depth in the weather file is missing or zero. Setting "
                                                "precipitation depth to 1.5 mm for this hour.",
                                                state.dataWaterData->PrecipOverwrittenByRainFlag);
             }
         }
+        if ((state.dataEnvrn->RunPeriodEnvironment) && (!state.dataGlobal->WarmupFlag)) {
+            int month = state.dataEnvrn->Month;
+            state.dataWaterData->RainFall.MonthlyTotalPrecInWeather.at(month - 1) += state.dataEnvrn->LiquidPrecipitation * 1000.0;
+            if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
+                state.dataWaterData->RainFall.numRainyHoursInWeather.at(month - 1) += 1;
+            }
+        }
 
         WaterManager::UpdatePrecipitation(state);
 
         state.dataEnvrn->TotalCloudCover = state.dataWeatherManager->TodayTotalSkyCover(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
         state.dataEnvrn->OpaqueCloudCover = state.dataWeatherManager->TodayOpaqueSkyCover(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
 
         if (state.dataWeatherManager->UseRainValues) {
-            state.dataEnvrn->IsRain = state.dataWeatherManager->TodayIsRain(
-                state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay); //.or. LiquidPrecipitation >= .8d0)  ! > .8 mm
+            state.dataEnvrn->IsRain = state.dataWaterData->RainFall.CurrentAmount > 0.0;
```

Comment:
Is it ever possible for the weather file data to have a non-zero (small) value for LiquidPrecipitation but TodayIsRain is false? If so, this will change old behavior and flip the rain flag to true.

Reply:
I'll double-check whether this could happen. The other behavior change would be if weather data has TodayIsRain = false but user-input site:precipitation defined non-zero rainfall value.

Reply:
>  The other behavior change would be if weather data has TodayIsRain = false but user-input site:precipitation defined non-zero rainfall value.
True, but that's the whole point of this PR in the first place. So that's a good change.

Reply:
I just checked the TMY3 data in the EnergyPlus/weather folder. All of them have PresWeathCodes = 999999999. As a result, none of them can set TodayIsRain to true. I'll download some more weather data and see.

Reply:
@mjwitte I also downloaded 36 more TMY3 whose states are not covered by the weather files in EnergyPlus folder. All of them also have PresWeathCodes = 999999999.